spec/models/comfy/cms/page_spec.rb

```diff
@@ -1,5 +1,78 @@
 RSpec.describe Comfy::Cms::Page do
 
+  describe '#related_links' do
+    let(:animals_tag) { Tag.create value: 'animals' }
+    let(:cheese_tag) { Tag.create value: 'cheese' }
+    let(:animals_article) { create :english_article, keywords: [animals_tag] }
+
+    context 'the more matching tags, the closer the match' do
+      context 'single keyword' do
+        it 'provides an empty array when no matching articles exist' do
+          expect(animals_article.related_links(1)).to be_empty
+        end
+
+        it 'provides one matching articles based on single keyword' do
+          related_animal_article = create :english_article, keywords: [animals_tag]
+          create :english_article, keywords: [cheese_tag]
+
+          expect(animals_article.related_links(2)).to eq([related_animal_article])
+        end
+      end
+
+      context 'two keywords' do
+        let(:dog_tag) { Tag.create value: 'dog' }
+        let!(:animals_article) { create :english_article, keywords: [animals_tag, dog_tag] }
+
+        it 'order by most matching tags' do
+          one_tag_match_article = create :english_article, keywords: [animals_tag]
+          all_tag_match_article = create :english_article, keywords: [animals_tag, dog_tag], page_views: 2
+
+          expect(animals_article.related_links(2)).to eq([all_tag_match_article, one_tag_match_article])
+        end
+      end
+    end
+
+    context 'page_view ordering matches' do
+      it 'orders by page_view' do
+        related_unpopular_article = create :english_article, keywords: [animals_tag], page_views: 1
+        related_popular_article = create :english_article, keywords: [animals_tag], page_views: 2
+
+        expect(animals_article.related_links(2)).to eq([related_popular_article, related_unpopular_article])
+      end
+    end
+
+    context 'when finding a welsh article' do
+      it 'we should exclude the alternate english version from the results' do
+        welsh_animal_article = create :welsh_article, keywords: [animals_tag], page_views: 2
+        related_article = create :english_article, keywords: [animals_tag], page_views: 1
+
+        allow(welsh_animal_article).to receive(:mirrors).and_return([animals_article])
```

Comment:
Related to the mirrors stuff, but this test passes if you comment out this line.